### DISCUSSION TITLE
Prediction with track identity

### DISCUSSION THREAD

**Discussion by Levans5** [talmolab/sleap#823] [Help!] [answered]
I am new to using SLEAP and currently working on a multi-animal project where I need 3 tracks. When I ran Inference on a 1000 frame clip for practice there are long instances of lost identity where a new track is generated for 50+ frames. Is it possible to select an animals inference in all 50 frames at once and switch its track to be the correct one?  Thanks.

**Comment by roomrys** [comment, marked as answer]
Hi @Levans5,

It is possible to switch the entire selected track via [proofreading](https://sleap.ai/develop/guides/proofreading.html#id1) (from the current frame to all subsequent frames - no previous frames will be switched):

1. Ensure "View" > "Color Predicted Instances" is on
2. Ensure "Tracks" > "Propagate Track Labels" is on
3. Go to the frame where the track switch initially occurs
4. Click on the instance you would like to switch the track of
![image](https://user-images.githubusercontent.com/38435167/176780069-1c524351-ae01-4f92-a76a-322fcbcaafd6.png)
**Fig 1:** Instance selected.

5. Hold `Ctrl` to see numeric track labels pop up (in upper left of GUI)
![image](https://user-images.githubusercontent.com/38435167/176780202-fdad7832-0b0e-4eea-83a2-7563669136eb.png)
**Fig 2:** Numeric track labels are listed in the parentheses.

6. While holding control, select the track number to switch to, i.e. `Ctrl` + `1`

Let us know if that is what you were looking for.

Thanks,
Liezl

**Reply by Levans5** [reply]
Hi Liezl

Thank you. I understand that this can be done one frame at a time. However, say an animal is assigned track-3 for 30 frames in a row is it possible to select those 30 frames for that instance and change the animal to be track-2 (the correct one). Or do I have to go through every single frame individually to proofread and make corrections.

**Reply by roomrys** [reply]
Hi @Levans5,

Selecting "Tracks" > "Propagate Track Labels" should propagate the new track label for the 30 frames in a row - you do **NOT** need to go through every single frame individually.

Thanks,
Liezl
